plugins/plugin-testing-java/plugin-testing-testng/che-plugin-testing-testng-server/src/main/java/org/eclipse/che/plugin/testing/testng/server/TestNGRunner.java

```diff
@@ -13,91 +13,73 @@
-import org.eclipse.jdt.core.ICompilationUnit;
-import org.eclipse.jdt.core.IJavaProject;
-import org.eclipse.jdt.core.IMethod;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.xml.sax.SAXException;
 
 /** TestNG implementation for the test runner service. */
 public class TestNGRunner extends AbstractJavaTestRunner {
   private static final String TESTNG_NAME = "testng";
   private static final Logger LOG = LoggerFactory.getLogger(TestNGRunner.class);
   private static final String JAVA_IO_TMPDIR = "java.io.tmpdir";
+  private static final String TEST_OUTPUT_FOLDER = "/test-output";
 
   private String workspacePath;
-  private JavaTestFinder javaTestFinder;
-  private final ProjectClasspathProvider classpathProvider;
+  private JavaLanguageServerExtensionManager extensionManager;
   private final TestNGSuiteUtil suiteUtil;
 
   @Inject
   public TestNGRunner(
       @Named("che.user.workspaces.storage") String workspacePath,
-      JavaTestFinder javaTestFinder,
-      ProjectClasspathProvider classpathProvider,
+      JavaLanguageServerExtensionManager extensionManager,
       TestNGSuiteUtil suiteUtil) {
-    super(workspacePath, javaTestFinder);
+    super(extensionManager);
     this.workspacePath = workspacePath;
-    this.javaTestFinder = javaTestFinder;
-    this.classpathProvider = classpathProvider;
+    this.extensionManager = extensionManager;
     this.suiteUtil = suiteUtil;
   }
 
   @Override
   @Nullable
   public ProcessHandler execute(TestExecutionContext context) {
-    IJavaProject javaProject = getJavaProject(context.getProjectPath());
-    if (javaProject.exists()) {
-      return startTestProcess(javaProject, context);
-    }
-
-    return null;
+    return startTestProcess(context);
   }
 
-  private ProcessHandler startTestProcess(IJavaProject javaProject, TestExecutionContext context) {
-    File suiteFile = createSuite(context, javaProject);
+  private ProcessHandler startTestProcess(TestExecutionContext context) {
+    File suiteFile = createSuite(context);
     if (suiteFile == null) {
       throw new RuntimeException("Can't create TestNG suite xml file.");
     }
 
     JavaParameters parameters = new JavaParameters();
     parameters.setJavaExecutable(System.getProperty("java.home") + "/bin/java");
     parameters.setMainClassName("org.testng.CheTestNGLauncher");
-    String outputDirectory = getOutputDirectory(javaProject);
+    String outputDirectory = getOutputDirectory(context);
     parameters.getParametersList().add("-d", outputDirectory);
-    parameters.setWorkingDirectory(workspacePath + javaProject.getPath());
+    parameters.setWorkingDirectory(workspacePath + context.getProjectPath());
     List<String> classPath = new ArrayList<>();
-    Set<String> projectClassPath = classpathProvider.getProjectClassPath(javaProject);
-    classPath.addAll(projectClassPath);
+    List<String> resolvedClassPaths = getResolvedClassPaths(context);
+    classPath.addAll(resolvedClassPaths);
```

Comment:
Personally, I think we should let the ExecutionException propagate from this method instead of returning null.